### ISSUE TITLE
consider adding tests directory to npmignore due to windows 260 character path limitation

### ISSUE THREAD

**Issue by dcbarans** [sass/node-sass#683]
As the title states. I've hit a wall where windows just falls on its face with paths longer than 260 characters. Here's an example path (not including the parent module its inside + my working directory path):
```
node-sass/test/fixtures/spec/spec/extend-tests/104_test_nested_extender_counts_extended_subselectors/104_test_nested_extender_counts_extended_subselectors'
```
The problem is compounded severely when node-sass is used as a module with other modules.

There's 2 solutions i see quickly.
* Add tests directory to npmigore. Though not sure if that's a path you'd be willing to explore.
* Shorten the naming convention of your test directories. Some are crazy long :)

**Comment by am11** [comment]
On Windows 8.1 x64, I keep node-sass under VS Git directory: `C:\Users\Adeel\Source\Repos\node-sass`. I don't have this issue. However, if we have further deep nesting this crazy limitation enforced by Windows will definitely present issues. There is only one true workaround for this issue: use longpath tool; since it is a restriction imposed by OS not NTFS itself (you may want to take a look at the `Pri.LongPath` workaround described here: https://github.com/npm/npm/issues/3697#issuecomment-62281110).

I am afraid we cannot do much about it, because spec is a submodule in our fixture and those file names belong to spec repo: https://github.com/sass/sass-spec.

**Comment by xzyfer** [comment]
@am11 do the npm install scripts still need the specs out of curiosity?

**Comment by am11** [comment]
@xzyfer, not really. With recent modifications, we now only execute *one* test on `npm install` (via mocha code usage) to verify if the binary is compatible and whether to fallback to manual build. What do you think if we (somehow) inline that test code and add `test/` to npmignore as suggested by @dcbarans?

**Comment by mgreter** [comment]
By the way, this is a windows limit as described at: http://stackoverflow.com/a/1880453/1550314

**Comment by xzyfer** [comment]
Presumably would could move the tests we want into the node-sass tests suite and `.npmignore` sass-spec. Seems like a waste for people to download sass-spec each time they npm install, and if it's causing legitimate issues we should ditch it.

It's worth pointing out that the directory depth problem in sass-spec is going to get worse before it gets better i.e. https://github.com/sass/sass-spec/pull/262/files

**Comment by am11** [comment]
I concur. We will discard `test/` directory from npm package and `mocha` will move down to dev-dependencies (since it will then be only required by the coverage script).

**Comment by dcbarans** [comment]
Thanks guys! I hit this error trying to install laravel-elixir which had node-sass nested 2 modules deep.. So i'm sure you will make many lives easier. Now to just get the Windows team to fix issues as quickly as you guys!

**Comment by am11** [comment]
This is fixed by 9b034c via #717.

**Comment by am11** [comment]
Turned out we don't need `.npmignore` for this. package.json would do the trick. See https://github.com/sass/node-sass/commit/fffc06259cc383f14218bb159b7bdbd6ce95d932#commitcomment-8369073.

Re-fixed by #724.
